testing/trino-faulttolerant-tests/src/test/java/io/trino/faulttolerant/BaseFailureRecoveryTest.java

```diff
@@ -310,6 +310,20 @@ public void testUpdateWithSubquery()
                 Optional.of("DROP TABLE <table>"));
     }
 
+    @Test(invocationCount = INVOCATION_COUNT)
+    public void testMerge()
+    {
+        testTableModification(
+                Optional.of("CREATE TABLE <table> AS SELECT * FROM orders"),
+                """
+                        MERGE INTO <table> t
+                        USING (SELECT orderkey, 'X' clerk FROM <table>) s
+                        ON t.orderkey = s.orderkey
+                        WHEN MATCHED THEN UPDATE SET clerk = t.clerk || s.clerk
```

Comment:
nit: not sure if that matters but maybe we should use MERGE statement which mixes UPDATES and DELETES here, just to have slightly better coverage in smoke test.

Reply:
WHEN clause changed to:

```
                        WHEN MATCHED AND s.orderkey > 1000
                            THEN UPDATE SET clerk = t.clerk || s.clerk
                        WHEN MATCHED AND s.orderkey <= 1000
                            THEN DELETE
```